components/popover/popover-mixin.js

```diff
@@ -100,19 +106,41 @@ export const PopoverMixin = superclass => class extends superclass {
 			this._previousFocusableAncestor = this.opened ? getPreviousFocusableAncestor(this, false, false) : null;
 
 			if (this.opened) {
+
 				this._opener = getComposedActiveElement();
 				this._addAutoCloseHandlers();
-				this._dismissibleId = setDismissible(() => this._close());
+				this._dismissibleId = setDismissible(() => this.close());
+				this._focusContent(this);
 				this.dispatchEvent(new CustomEvent('d2l-popover-open', { bubbles: true, composed: true }));
+
 			} else if (changedProperties.get('opened') !== undefined) {
+
 				this._removeAutoCloseHandlers();
 				this._clearDismissible();
+				this._focusOpener();
 				this.dispatchEvent(new CustomEvent('d2l-popover-close', { bubbles: true, composed: true }));
+
 			}
 
 		}
 	}
 
+	close() {
```

Comment:
Should this (and `open` and `toggleOpen`) be `async` since it's returning a Promise?

Reply:
Na, it's sorta redundant if we apply `async`, at least in this case. The only reason to use async in this case would be if we wanted to internally `await` some other Promise, or if we feel strongly about hiding the result of the internal Promise.

Three cases, all return a Promise...

Obviously returns Lit's Promise:
```javascript
close() {
    ...
    return this.updateComplete;
}
```

Also returns Lit's Promise (the browser sees the Promise that is `return`'d, and returns _that_):
```javascript
async close() {
    ...
    return this.updateComplete;
}
```

Returns a new Promise, effectively keeping Lit's Promise internal.
```javascript
async close() {
    ...
    await this.updateComplete;
}
```

Reply:
Right. I guess I was thinking more about consumers that might want to `await` the methods before proceeding with something else, like a vdiff test or whatever.